src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,88 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] baskets;
+
+    public MyHashMap() {
+        baskets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        Node<K,V> newNode = new Node<>(key, value, null);
+        if (baskets[index] == null) {
+            baskets[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K,V> oldNode = baskets[index];
+        if (oldNode.key == key || oldNode.key != null && oldNode.key.equals(key)) {
+            oldNode.value = value;
+            return;
+        }
+        while (oldNode.next != null) {
+            if (oldNode.key == key || oldNode.key != null && oldNode.key.equals(key)) {
+                oldNode.value = value;
+                return;
+            }
+            oldNode = oldNode.next;
+        }
+        oldNode.next = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = baskets[index];
+        while (currentNode != null) {
+            if (currentNode.key == key
+                    || currentNode.key != null
+                    && currentNode.key.equals(key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        if (size >= baskets.length * DEFAULT_LOAD_FACTOR) {
+            Node<K, V>[] oldBaskets = baskets;
+            baskets = new Node[baskets.length * 2];
+            size = 0;
+            for (Node<K, V> node : oldBaskets) {
+                while (node != null) {
+                    put(node.key, node.value);
+                    node = node.next;
+                }
+            }
+        }
+    }
+
+    private int getIndex(Object key) {
+        return key == null ? 0 : (key.hashCode() & baskets.length - 1);
+    }
+
+    private static class Node<K, V> {
+        K key;
+        V value;
+        Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
```

Comment:
```suggestion
        Node(K key, V value) {
```
you create node only in one place, where you don't use third input value, so let' remove it

Reply:
done